go/service/usersearch.go

```diff
@@ -167,24 +173,23 @@ func contactSearch(mctx libkb.MetaContext, arg keybase1.UserSearchArg) (res []ke
 	}
 
 	for _, c := range contactsRes {
-		found, score := matchAndScoreContact(query, c)
+		found, score, matchedVal := matchAndScoreContact(query, c)
 		if found {
 			contact := c
-			contact.RawScore = score
+			if contact.Resolved && matchedVal != "" {
+				// If contact is resolved, make sure to plumb matched query to
+				// display label. This is not needed for unresolved contacts,
+				// which can only match on ContactName or Component Value, and
+				// both of them always appear as name and label.
+				contact.DisplayLabel = matchedVal
```

Comment:
We talked about formatting returned phone numbers here, so we don't have to do a bunch of "isLabelPhoneNumber" and "isNamePhoneNumber" branching logic in the frontend. Is that still the plan?

Reply:
It's still part of the grand plan. There are few issues with current design, e.g. it should not do the deduplication during contact syncing stage, but rather save everything and only process that when fetching contacts.

I wanted this PR to be a simple fix to most apparent issues.